File: JavaScript_/BinaryTree/bfs.js
class Node{
  constructor(val){
    this.val =  val;
    this.left = null;
    this.right= null;
  }
  
}

const breadthFirstValue = (root) =>{
  if (root === null) return [];
  
  const values = []
  const queue = [root];
  
  
  while (queue.length > 0){
    const current = queue.shift();
    values.push(current.val);
    
    if (current.left != null) queue.push(current.left);
    if (current.right != null) queue.push(current.right);
    
    
  }
  
  return values
  
}


const a = new Node("A")
const b = new Node("B")
const c = new Node("C")
const d = new Node("D")
const e = new Node("E")
const f = new Node("F")

a.left = b;
a.right = c;
b.left = d ;
b.right = e ;
c.right = f;

// #     a
// #    /  \
// #    b  c
// #  /  \  \
// #  d  e   f

console.log(
  breadthFirstValue(a)
)

File: Python_/Binary_tree/bfs.py
from collections import deque


class Node:
    def __init__(self, value):
        self.val = value
        self.left = None
        self.right = None


def breadth_first_values(root):
    if not root:
        return []

    queue = deque([root])
    values = []

    while queue:
        node = queue.popleft()

        values.append(node.val)

        if node.left:
            queue.append(node.left)

        if node.right:
            queue.append(node.right)

    return values


root = Node(1)
root.left = Node(2)
root.right = Node(3)

root.left.left = Node(4)
root.left.right = Node(5)

root.right.left = Node(6)
root.right.right = Node(7)


#     1
#    / \
#   2   3
#  / \ / \
# 4  5 6  7

print(breadth_first_values(root))
